app/components/comment-item.js

```diff
@@ -1,54 +1,155 @@
+const {
+  Component,
+  computed,
+  get,
+  inject,
+  isEqual,
+  set,
+} = Ember;
+
+const { service } = inject;
+
+/**
+  `comment-item` composes a comment
+
+  ```handlebars
+  {{comment-item comment=comment}}
+  ```
+
+  @class comment-item
+  @module Component
+  @extends Ember.Component
+ */
+export default Component.extend({
   classNames: ['comment-item', 'timeline-comment-wrapper'],
   classNameBindings: ['isEditing:editing'],
 
-  currentUser: Ember.inject.service(),
-  mentionFetcher: Ember.inject.service(),
+  /**
+    @property currentUser
+    @type Ember.Service
+   */
+  currentUser: service(),
+
+  /**
+    A service that is used for fetching mentions within a body of text.
+
+    @property mentionFetcher
+    @type Ember.Service
+   */
+  mentionFetcher: service(),
+
+  /**
+    Returns whether or not the current user can edit the current comment.
+
+    @property canEdit
+    @type Boolean
+   */
+  canEdit: alias('currentUserIsCommentAuthor'),
+
+  /**
+    Returns the comment author's ID.
+
+    @property commentAuthorId
+    @type Number
+   */
+  commentAuthorId: alias('comment.user.id'),
+
+  /**
+    Returns the current user's ID.
 
-  canEdit: Ember.computed.alias('currentUserIsCommentAuthor'),
-  commentAuthorId: Ember.computed.alias('comment.user.id'),
-  currentUserId: Ember.computed.alias('currentUser.user.id'),
-  currentUserIsCommentAuthor: Ember.computed('currentUserId', 'commentAuthorId', function() {
-    let userId = parseInt(this.get('currentUserId'), 10);
-    let authorId = parseInt(this.get('commentAuthorId'), 10);
-    return userId === authorId;
+    @property currentUserId
+    @type Number
+   */
+  currentUserId: alias('currentUser.user.id'),
+
+  /**
+    Consumes `currentUserId` and `commentAuthorId` and returns if the current
+    user is the comment author.
+
+    @property currentUserIsCommentAuthor
+    @type Boolean
+   */
+  currentUserIsCommentAuthor: computed('currentUserId', 'commentAuthorId', function() {
+    let userId = parseInt(get(this, 'currentUserId'), 10);
+    let authorId = parseInt(get(this, 'commentAuthorId'), 10);
+
+    return isEqual(userId, authorId);
   }),
 
+  /**
+    Overrides the init method, sets the `isEditing` property to `false`,
+    prefetches menthions and calls the `super` on `init`.
+
+    @method init
+   */
   init() {
-    this.set('isEditing', false);
-    this._prefetchMentions(this.get('comment'));
+    set(this, 'isEditing', false);
+    this._prefetchMentions(get(this, 'comment'));
     return this._super(...arguments);
   },
 
   actions: {
+
+    /**
+      Action that stops the editing of the corresponding comment.
+
+      @method cancel
+     */
     cancel() {
-      this.set('isEditing', false);
+      set(this, 'isEditing', false);
     },
 
+    /**
+      Action that sets the corresponding comment to edit mode.
+
+      @method edit
+     */
     edit() {
-      this.set('isEditing', true);
+      set(this, 'isEditing', true);
```

Comment:
This doesnt really have to do with the docs, but couldnt we condense the cancel and edit actions to be just one toggle action?
```
toggleEdit() {
  this.toggleProperty('isEditing');
}
```